src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/SpecialFileProviders/AbstractSpecialFileProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.SpecialFileProviders
+{
+    /// <summary>
+    ///     Provides the base class for all <see cref="ISpecialFileProvider"/> instances.
+    /// </summary>
+    internal abstract class AbstractSpecialFileProvider : ISpecialFileProvider
+    {
+        private readonly IProjectTreeService _treeService;
+
+        protected AbstractSpecialFileProvider(IProjectTreeService treeService)
+        {
+            _treeService = treeService;
+        }
+
+        public virtual async Task<string?> GetFileAsync(SpecialFiles fileId, SpecialFileFlags flags, CancellationToken cancellationToken = default)
```

Comment:
MEF Metadata on the type tells CPS what special file the handler handles - it could be many, ours are 1 to 1, though. We could add a field to check this, but it will just restate what the metadata states and would solely be for an assert or argument throw that would never occur.